integration-libs/cds/src/components/recent-searches/recent-searches.service.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { interval, last, of, ReplaySubject } from 'rxjs';
+import { concatMap, map, take, takeWhile } from 'rxjs/operators';
+import { WindowRef } from '@spartacus/core';
+
+@Injectable({ providedIn: 'root' })
+export class RecentSearchesService {
+  private readonly recentSearchesSource = new ReplaySubject<string[]>();
+  public recentSearches$ = this.recentSearchesSource.asObservable();
+  constructor(protected winRef: WindowRef) {
+    this.addRecentSearchesListener();
+  }
+
+  checkAvailability() {
```

Comment:
I'm wondering whether this function should be public. Is it utilized anywhere beyond this service?